vivisect/parsers/elf.py

```diff
@@ -194,10 +200,48 @@ def loadElfIntoWorkspace(vw, elf, filename=None):
             vw.makeName(sva, "init_function", filelocal=True)
             vw.addEntryPoint(sva)
 
+        elif sname == ".init_array":
+            # handle pseudo-fixups first: these pointers require base-addresses
+            psize = vw.getPointerSize()
+            pfmt = e_bits.le_fmt_chars[psize]   #FIXME: make Endian-aware (needs plumbing through ELF)
+            secbytes = elf.readAtRva(sec.sh_addr, size)
+            sh_addr = sec.sh_addr
+            if addbase: sh_addr += baseaddr
+
+            ptr_count = 0
+            for off in range(0, size, psize):
+                addr, = struct.unpack_from(pfmt, secbytes, off)
+                if addbase: addr += baseaddr
+                
+                vw.makePointer(sh_addr + off, addr)
+                vw.makeName(addr, "init_function_%d" % ptr_count, filelocal=True)
+                vw.addXref(sh_addr + off, addr, REF_PTR)
+                vw.addEntryPoint(addr)
+                ptr_count += 1
```

Comment:
In terms of cleanliness, might be good to carve this .init_array and .fini_array conditional sections out into a common function, since it seems like you're doing exactly the same things here, save for the names.